File: Two Sum (Easy, #1)/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        size = len(nums)
        nums_dict = dict()
        for i in range(size):
            nums_dict[nums[i]] = i

        index = 0;        
        for i in nums:
            diff = target - i
            if diff in nums_dict and index != nums_dict[diff]:
                return [index, nums_dict[diff]]
            index += 1
    
# This solution has time complexity of O(n) and space complexity of O(n)

File: Two Sum (Easy, #1)/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int nums_length = nums.size();
        for(int i = 0; i < nums_length; ++i) {
            int diff = target - nums[i];
            for (int j = i + 1; j < nums_length; ++j) {
                if(nums[j] == diff)
                    return vector<int>{i, j};
            }
        }
        return vector<int>{};
    }
};

// Currently, such a solution is of O(n^2) of time complexity, and O(1) of space complexity. 
// To make this solution faster, we can use an unordered map with the keys being the values in the vector nums and the values being
// the indexes of those values. I will implement this in python so that the time complexity is O(n) and the space complexity is O(n)